be/src/vec/aggregate_functions/aggregate_function_group_array_set_op.h

```diff
@@ -0,0 +1,483 @@
+public:
+    NullableStringSet() : StringValueSet<DynamicContainer<StringRef>>(true) {}
+
+    void change_contain_null_value(bool target_value) { this->_contain_null = target_value; }
+};
+
+struct GroupArraySetOpStringBaseData {
+    using Set = std::unique_ptr<NullableStringSet>;
+
+    GroupArraySetOpStringBaseData() : set(std::make_unique<NullableStringSet>()) {}
+    Set set;
+    bool init = false;
+
+    void reset() {
+        init = false;
+        set = std::make_unique<NullableStringSet>();
+    }
+
+    void serialize(BufferWritable& buf) const {
+        const bool is_set_contain_null = this->set->contain_null();
+        buf.write_binary(is_set_contain_null);
+        buf.write_binary(this->init);
+        buf.write_var_uint(this->set->size());
+
+        HybridSetBase::IteratorBase* it = this->set->begin();
+        while (it->has_next()) {
+            const auto* value = reinterpret_cast<const StringRef*>(it->get_value());
+            buf.write_binary(*value);
+            it->next();
+        }
+    }
+
+    void deserialize(BufferReadable& buf, Arena& arena) {
+        bool is_set_contain_null;
+        buf.read_binary(is_set_contain_null);
+        this->set->change_contain_null_value(is_set_contain_null);
+        buf.read_binary(this->init);
+        UInt64 size;
+        buf.read_var_uint(size);
+
+        StringRef element;
+        for (UInt64 i = 0; i < size; ++i) {
+            element = buf.read_binary_into(arena);
+            this->set->insert((void*)element.data, element.size);
+        }
+    }
+
+    void insert_result_into(IColumn& to) const {
+        auto& arr_to = assert_cast<ColumnArray&>(to);
+        ColumnArray::Offsets64& offsets_to = arr_to.get_offsets();
+        auto& data_to = arr_to.get_data();
+        auto* col_null = assert_cast<ColumnNullable*>(&data_to);
+        auto res_size = this->set->size();
+
+        if (this->set->contain_null()) {
+            col_null->insert_default();
+            res_size += 1;
+        }
+
+        offsets_to.push_back(offsets_to.back() + res_size);
+        HybridSetBase::IteratorBase* it = this->set->begin();
+        while (it->has_next()) {
+            const auto* value = reinterpret_cast<const StringRef*>(it->get_value());
+            data_to.insert_data(value->data, value->size);
+            it->next();
+        }
+    }
+};
+
+struct GroupArrayStringIntersectData : public GroupArraySetOpStringBaseData {
+    using Base = GroupArraySetOpStringBaseData;
+    using Set = Base::Set;
+
+    static std::string get_name() { return "group_array_intersect"; }
+
+    void process_col_data(const auto& column_data, size_t offset, size_t arr_size, Arena& arena) {
+        const auto* col_null = assert_cast<const ColumnNullable*>(column_data.get());
+        const auto& nested_column_data =
+                assert_cast<const ColumnString&>(col_null->get_nested_column());
+        auto process_element = [&](size_t i) {
```

Comment:
maybe we should not use area to save all element in array. which may cause mem release later. the op occupy to many mem